File: Lecture_2/readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");
    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {

        if (text[i] != ' ')
        {
            if (text[i] >= 'a' && text[i] <= 'z')
            {
                letters++;
            }
            else if (text[i] >= 'A' && text[i] <= 'Z')
            {
                letters++;
            }
        }
        if (text[i] == ' ')
        {
            words++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8;
    int rIndex = round(index);

    if (rIndex < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (rIndex > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", rIndex);
    }
}

File: Lecture_6/sentimental-readability/readability.py
text = input("Text: ").lower()

letters = 0
words = 1
sentences = 0

for i in text:
    if i != ' ':
        if i >= 'a' and i <= 'z':
            letters += 1
    if i == ' ':
        words += 1
    if i == '.' or i == '!' or i == '?':
        sentences += 1

L = float(letters / words) * 100
S = float(sentences / words * 100)

index = 0.0588 * L - 0.296 * S - 15.8
rIndex = int(round(index))

print(rIndex)
if rIndex < 1:
    print("Before Grade 1")
elif rIndex > 16:
    print("Grade 16+")
else:
    print(f"Grade {rIndex}")
